src/m1_facilitator/SeedDataInitializer.ts

```diff
@@ -0,0 +1,197 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Web3 from 'web3';
+import * as Mosaic from 'Mosaic';
+import BigNumber from 'bignumber.js';
+
+import Gateway, { GatewayType } from './models/Gateway';
+import Anchor from './models/Anchor';
+import Repositories from './repositories/Repositories';
+import ContractEntity, { EntityType } from '../common/models/ContractEntity';
+import Utils from '../common/Utils';
+
+export default class SeedDataInitializer {
+  /** Instance of Repositories class. */
+  private repositories: Repositories;
+
+  /**
+   * Construct the SeedDataInitializer with the params.
+   *
+   * @param repositories Instance of repository class.
+   */
+  public constructor(
+    repositories: Repositories,
+  ) {
+    this.repositories = repositories;
+  }
+
+  /**
+   * Generate and save records that should be populated as seed data.
+   * - Save ERC20Gateway record
+   * - Save ERC20goGateway record
+   * - Save Origin anchor record
+   * - Save auxiliary anchor record
+   * - Save contract entity records.
+   *
+   * @param originWeb3 Instance of origin web3.
+   * @param auxiliaryWeb3 Instance of auxiliary web3.
+   * @param gatewayAddresses Gateway address.
+   */
+  public async initialize(
+    originWeb3: Web3,
+    auxiliaryWeb3: Web3,
+    gatewayAddresses: string,
+  ): Promise<void> {
+    const erc20Gateway = Mosaic.interacts.getERC20Gateway(originWeb3, gatewayAddresses);
+    const cogatewayAddress = await erc20Gateway.methods.messageOutbox().call();
+    const erc20Cogateway = Mosaic.interacts.getERC20Cogateway(
+      auxiliaryWeb3,
+      cogatewayAddress,
+    );
+
+    const originAnchorAddress = await erc20Gateway.methods.stateRootProvider().call();
+    const originAnchorInstance = Mosaic.interacts.getAnchor(
+      originWeb3,
+      originAnchorAddress,
+    );
+
+    const auxiliaryAnchorAddress = await erc20Cogateway.methods.stateRootProvider().call();
+    const auxiliaryAnchorInstance = Mosaic.interacts.getAnchor(
+      auxiliaryWeb3,
+      auxiliaryAnchorAddress,
+    );
+
+    const originLatestAnchoredStateRootBlockHeight = await originAnchorInstance.methods
+      .getLatestStateRootBlockNumber().call();
+    const auxiliaryLatestAnchoredStateRootBlockHeight = await auxiliaryAnchorInstance.methods
+      .getLatestStateRootBlockNumber().call();
+
+    const originGateway = new Gateway(
+      Gateway.getGlobalAddress(gatewayAddresses),
+      Gateway.getGlobalAddress(cogatewayAddress),
+      GatewayType.ERC20,
+      Anchor.getGlobalAddress(originAnchorAddress),
+      new BigNumber(0),
+    );
+
+    const auxiliaryGateway = new Gateway(
+      Gateway.getGlobalAddress(cogatewayAddress),
+      Gateway.getGlobalAddress(gatewayAddresses),
+      GatewayType.ERC20,
+      Anchor.getGlobalAddress(auxiliaryAnchorAddress),
+      new BigNumber(0),
```

Comment:
Why do we assume that remoteGatewayLastProvenBlockNumber is 0?

Reply:
It should not be, it should fetch lastestGatewayProvenBlockHeight from contract, but currently there is no method to get this value. We need to create ticket for this.

Reply:
We will create a function on contract to get the `latestGatewayProvenBlockHeight` form contract.